src/transforms/kubernetes/mod.rs

```diff
@@ -67,12 +102,17 @@ impl TransformConfig for KubePodMetadata {
 
         // Run background task
         cx.executor().spawn_std(async move {
-            let error = metadata_client.run().await;
-            error!(message = "Stopped updating Pod metadata.", reason = %error);
+            match metadata_client.run().await {
+                Ok(()) => unreachable!(),
```

Comment:
I would think we probably just want to let this pass through, since technically a task could finish gracefully and just exit? (even if it doesn't work like this now thats how I would assume this type of api would work.

Reply:
True.